app/bot/openai.go

```diff
@@ -52,11 +52,11 @@ func (o *OpenAI) OnMessage(msg Message) (response Response) {
 		return Response{}
 	}
 
-	if o.nowFn().Sub(o.lastDT) < 30*time.Minute && !o.superUser.IsSuper(msg.From.Username) {
-		log.Printf("[WARN] OpenAI bot is too busy, last request was %s ago, %s banned", time.Since(o.lastDT), msg.From.Username)
+	banMessage := ""
+	ok, banMessage = o.checkRequest(msg.From.Username, reqText)
```

Comment:
a small thing - `if ok, banMessage = func() {` would make it shorter and more readable